### PR TITLE
Scene without arcs

### PR DESCRIPTION
**Pull request by wkozyra95**
7wwwjy

### CODE REVIEW

File: compositor_render/src/renderer/scene.rs [hunk 1]
```diff
@@ -207,40 +216,94 @@ impl Scene {
...
+        let node = node_and_inputs
+            .remove(&node_id)
+            .ok_or_else(|| SceneError::MissingNode(node_id.clone()))?;
+        let inputs: Vec<&Node> = input_ids
+            .into_iter()
+            .map(|input_id| {
+                node_and_inputs
+                    .remove(&input_id)
+                    .ok_or_else(|| SceneError::MissingNode(node_id.clone()))
+                    .map(|r| &*r)
```
Comment (wkozyra95) [inline compositor_render/src/renderer/scene.rs:292]: not necessary, but I wanted to explicitly cast down &mut Node to &Node for inputs

**Review by jerzywilczek — APPROVED** [review]
:crab:

File: compositor_render/src/renderer/scene.rs [hunk 2]
```diff
@@ -207,40 +216,108 @@ impl Scene {
...
+            .collect::<Result<Vec<_>, SceneError>>()?;
+        Ok(NodeRenderPass { node, inputs })
+    }
+}
+
+impl Default for SceneNodesSet {
+    fn default() -> Self {
+        Self::new()
+    }
+}
```
Comment (WojciechBarczynski) [inline compositor_render/src/renderer/scene.rs:317]: [NIT]
Can't this be derived?

**Review by WojciechBarczynski — APPROVED** [review]
GJ

**Pull request merged**